examples/bri-3/src/bri/zeroKnowledgeProof/agents/ccsmAnchorHash.agent.ts

```diff
@@ -0,0 +1,49 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { INVALID_ANCHOR_HASH_INPUT } from '../api/err.messages';
+import { v4 as uuidv4 } from 'uuid';
+import { CCSMAnchorHash } from '../models/ccsmAnchorHash';
+
+@Injectable()
+export class CCSMAnchorHashAgent {
+  public throwErrorIfCCSMAnchorHashInputInvalid(
+    inputForProofVerification: string,
+  ): void {
+    if (!inputForProofVerification) {
+      throw new BadRequestException(INVALID_ANCHOR_HASH_INPUT);
+    }
+  }
+
+  public createNewCCSMAnchorHash(
+    ownerId: string,
+    document: string,
+  ): CCSMAnchorHash {
+    const hash = this.convertDocumentToHash(document);
+
+    return new CCSMAnchorHash(uuidv4(), ownerId, hash);
+  }
+
+  public verifyCCSMAnchorHash(
+    CCSMAnchorHash: string,
+    publicInputForProofVerification: string,
+  ): boolean {
+    if (CCSMAnchorHash === publicInputForProofVerification) {
+      return true;
+    }
+
+    return false;
+  }
+
+  public createPublicInputForProofVerification(
```

Comment:
@biscuitdey again, unclear what that means until the prover system is specified. Please, specify all required elements of the prover system and the system for content addressable hash management (CRUD)

Reply:
in an issue of course

Reply:
I have created a separate issue for the CRUD of content addressable hash.